### PR TITLE
Migrate select team to redux store

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
While working in this PR: https://github.com/mattermost/mattermost-webapp/pull/1070, I thought of migrating this SelectItem to use our redux store. Also added test.

(Note: I'll just rebase after the said PR is accepted and merged.)

#### Ticket Link
none

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by saturninoabril** [comment]
I'm just making a few change, then I'll change the label for dev review.

**Comment by saturninoabril** [comment]
@jwilander @jespino Done rebase, fix conflicts and add tests. Should be ready for your review.

File: components/select_team/index.js [hunk 1]
```diff
@@ -9,17 +9,37 @@ import {getTeams} from 'mattermost-redux/actions/teams';
...
+        const options = {
+            numeric: true,
+            sensitivity: 'base',
+        };
+        return a.display_name.localeCompare(b.display_name, currentUser.locale || 'en', options);
+    }
+
+    const joinableTeams = Object.values(getJoinableTeams(state)).
+        filter((team) => team.delete_at === 0).
+        sort(sortTeams);
```
Comment (jwilander) [inline components/select_team/index.js]: This is going to be a new object every time, probably better to make a selector to do this

Comment (saturninoabril) [reply components/select_team/index.js]: Thanks!  I added selector for that.

File: components/select_team/index.js [hunk 2]
```diff
@@ -4,22 +4,49 @@
...
 import {getTeams} from 'mattermost-redux/actions/teams';
 import {loadRolesIfNeeded} from 'mattermost-redux/actions/roles';
 import {getConfig, getLicense} from 'mattermost-redux/selectors/entities/general';
 import {getRoles} from 'mattermost-redux/selectors/entities/roles';
+import {getJoinableTeams, getTeamMemberships} from 'mattermost-redux/selectors/entities/teams';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
 
 import SelectTeam from './select_team.jsx';
 
+const getSortedJoinableTeams = createSelector(
```
Comment (jespino) [inline components/select_team/index.js]: Maybe it makes sense to move it to mattermost-webapp/selectors, or mattermost-redux. But I'm 1/5 here.

Comment (saturninoabril) [reply components/select_team/index.js]: PR submitted - https://github.com/mattermost/mattermost-redux/pull/471

**Comment by saturninoabril** [comment]
Change on redux has been merged. This is now ready for review.

**Pull request merged**